linux/keyman-config/keyman_config/gsettings.py

```diff
@@ -121,9 +123,11 @@ def _set_key_file(self, key, value, type_string):
     def set(self, key, value, type_string) -> None:
         if self.is_sudo:
             variant = str(value)
-            subprocess.run(
-              ['sudo', '-H', '-u', os.environ.get('SUDO_USER'),
-               'gsettings', 'set', self.schema_id, key, variant], check=False)
+            args = ['sudo', '-H', '-u', os.environ.get('SUDO_USER'),
+                    f'DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/{os.environ.get("SUDO_UID")}/bus',
+                    f'XDG_RUNTIME_DIR=/run/user/{os.environ.get("SUDO_UID")}',
```

Comment:
These values are repeated and are complicated format strings; would be good to refactor them in gsettings.py and the similar one in dbus_util.py